libnavigation-core/src/main/java/com/mapbox/navigation/core/navigator/LocationEx.kt

```diff
@@ -48,11 +51,48 @@ internal fun Location.toFixLocation(): FixLocation {
         provider,
         bearingAccuracy,
         speedAccuracy,
-        verticalAccuracy
+        verticalAccuracy,
+        if (extras != null) extras.toMap() else Bundle().toMap()
     )
 }
 
 internal fun List<FixLocation>.toLocations(): List<Location> = this.map { it.toLocation() }
 
 private fun isCurrentSdkVersionEqualOrGreaterThan(sdkCode: Int): Boolean =
     Build.VERSION.SDK_INT >= sdkCode
+
+internal fun HashMap<String, Value>.toBundle(): Bundle? {
```

Comment:
I'm not sure that this is a good place to have extension functions for Bundle and HashMap inside Location.Ex.kt

Reply:
Let's keep it for now, it's `internal` and as I mentioned it'd be great to have these exposed https://github.com/mapbox/mapbox-navigation-android/pull/5034#discussion_r747684218